src/index.js

```diff
@@ -1,36 +1,150 @@
+    </div>
+  </li>`;
+      return listElement;
+    }).join('');
+    taskListContainer.insertAdjacentHTML('beforeend', elements);
+    const trashIcons = taskListContainer.querySelectorAll('.fa-trash-can');
+    const verticalIcons = taskListContainer.querySelectorAll(
+      '.fa-ellipsis-vertical',
+    );
+    const checkBoxes = taskListContainer.querySelectorAll('.check');
 
-window.addEventListener('DOMContentLoaded', populateToDoList);
+    verticalIcons.forEach((icon) => {
+      icon.addEventListener('click', (e) => {
+        const { Option } = e.target.dataset;
+        const parent = e.target.parentElement.parentElement;
+        const editInput = parent.querySelector('.edit--input');
+        const descPara = parent.querySelector('.desc');
+        const editForm = parent.querySelector('.edit-form');
+        parent.classList.add('hide--desc');
+        editInput.value = descPara.textContent;
+        editInput.focus();
+        editForm.addEventListener('submit', (e) => {
+          e.preventDefault();
+          if (editInput.value) {
+            this.editTask(Option, editInput.value);
+            this.populateToDoList();
+            this.setLocalStorage(this.myToDoList);
+          }
+        });
+      });
+    });
+    trashIcons.forEach((icon) => {
+      icon.addEventListener('click', (e) => {
+        const { Trash } = e.target.dataset;
+        this.removeTask(Trash);
+        this.updateTaskIndex();
+        this.populateToDoList();
+        this.setLocalStorage(this.myToDoList);
+      });
+    });
+    checkBoxes.forEach((checkBox) => {
+      checkBox.addEventListener('change', (e) => {
+        const { id } = e.target;
+        const parent = e.target.parentElement.parentElement;
+        const pTag = parent.querySelector('.desc');
+        pTag.classList.toggle('strike--through');
+        this.updateCompletionStatus(+id);
+        this.setLocalStorage(this.myToDoList);
+      });
+    });
+  };
+
+  addTask = (description, completed = false) => {
+    const index = this.myToDoList.length + 1;
+    const newTask = { description, completed, index };
+    this.myToDoList = [...this.myToDoList, newTask];
+  };
+
+  removeTask = (index) => {
+    this.myToDoList = this.myToDoList.filter((list) => list.index !== +index);
+  };
+
+  updateTaskIndex = () => {
+    this.myToDoList = this.myToDoList.map((list, index) => {
+      list.index = index + 1;
+      return list;
+    });
+  }
+
+  editTask = (index, value) => {
+    this.myToDoList = this.myToDoList.map((list) => {
+      if (list.index === +index) {
+        list.description = value;
+      }
+      return list;
+    });
+  };
+
+  updateCompletionStatus = (index) => {
+    this.myToDoList = this.myToDoList.map((list) => {
+      if (list.index === index) {
+        list.completed = !list.completed;
+      }
+      return list;
+    });
+  };
+
+  clearAllCompletedTasks = () => {
+    this.myToDoList = this.myToDoList.filter((list) => !list.completed);
+  };
```

Comment:
- [ ] Its a requirement that we should create a new JavaScript file to hold the functionality for this milestone. Kindly do so.Check the screenshot of the requirements below
![image](https://github.com/JoyTaribagshaw/To-Do-List-Project/assets/81932478/f72ce1b8-5d94-40c3-87ff-36497e251218)


- [ ] Good job implementing the task object's values for `completed` and adding a function that deletes all completed tasks. However, we have some issues we need to fix. The user is unable to add and delete tasks to the app, check the screencast below. Kindly fix this issue.

https://github.com/JoyTaribagshaw/To-Do-List-Project/assets/81932478/5cd45ccd-aebd-4fa5-a14d-197c84486a22

Reply:
Thank you so much for your review @Munsa1 , I have resolved the add and delete tasks functionality.